### PR TITLE
Add some options to dump the message expected by the Test Harness

### PR DESCRIPTION
**Pull request by vivien-apple**
#### Problem

In order to use the python runner, the test harness expects some special strings.
This is an attempt to get it correct.

### CODE REVIEW

**Comment by vivien-apple** [comment]
I have updated the PR to contains logs for the logs command as well as some logs when the device connection succeed or fails.

File: scripts/py_matter_yamltests/matter_yamltests/hooks.py [hunk 1]
```diff
@@ -126,14 +129,17 @@ def test_stop(self, exception: Exception, duration: int):
...
         """
         This method is called when running a step is skipped.
 
         Parameters
         ----------
         name: str
             The name of the test step that is skipped.
+
+        expression: str
+            The PICS expression that results in the test step to be skipped.
```
Comment (bzbarsky-apple) [inline scripts/py_matter_yamltests/matter_yamltests/hooks.py:142]:
```suggestion
            The PICS expression that results in the test step being skipped.
```

File: scripts/tests/yaml/runner.py [hunk 2]
```diff
@@ -71,6 +71,8 @@ def test_runner_options(f):
                      help='Show additional logs provided by the adapter.')(f)
     f = click.option('--show_adapter_logs_on_error', type=bool, default=True, show_default=True,
                      help='Show additional logs provided by the adapter on error.')(f)
+    f = click.option('--use_test_harness_log_format', type=bool, default=False, show_default=True,
+                     help='Use the test harness log format..')(f)
```
Comment (bzbarsky-apple) [inline scripts/tests/yaml/runner.py:75]:
```suggestion
                     help='Use the test harness log format.')(f)
```

**Review by bzbarsky-apple — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Comment by mikaelhm** [comment]
I verified with a PoC integration with Matter TH that this does work.

Screenshot 2023-04-28 at 8 56 52 AM

**Comment by mikaelhm** [comment]
@vivien-apple these are the strings TH is currently scanning for in `chip-tool` and `chip-app` log output:

```

#chip-tool test flow
"***** Test Step"
"***** Test Failure:"
"***** Step Failure:"
"**** Skipping:"

#chip-tool user-prompt matches:
"USER_PROMPT"

#chip-app - prompt tester to perform action
"Wait for"

#chip-app failure:
"AttributePath does not match"

#chip-app manual pairing code:
"Manual pairing code:"
```

File: scripts/tests/yaml/runner.py [hunk 3]
```diff
@@ -276,11 +278,11 @@ def run(parser_group: ParserGroup, adapter: str, stop_on_error: bool, stop_on_wa
 @test_runner_options
 @websocket_runner_options
 @pass_parser_group
-def websocket(parser_group: ParserGroup, adapter: str, stop_on_error: bool, stop_on_warning: bool, stop_at_number: int, show_adapter_logs: bool, show_adapter_logs_on_error: bool, server_address: str, server_port: int, server_path: str, server_name: str, server_arguments: str):
+def websocket(parser_group: ParserGroup, adapter: str, stop_on_error: bool, stop_on_warning: bool, stop_at_number: int, show_adapter_logs: bool, show_adapter_logs_on_error: bool, use_test_harness_log_format: bool, server_address: str, server_port: int, server_path: str, server_name: str, server_arguments: str):
```
[the remark this answers is not in the document]
Comment (vivien-apple) [reply scripts/tests/yaml/runner.py:281]: > This is now a method with 12 arguments...could we move some of them to a dataclass with builders to make it easier to read? Same comment for the rest ... anything over say 5 is not really maintainable.

That seems unrelated to this very particular PR. But I am adding that to my list.

**Pull request merged**